recipe/recipe.yaml

```diff
@@ -0,0 +1,227 @@
+
+schema_version: 1
+
+context:
+  version: 2.3.1
+  # -1 means "not a dev version", put in a string like ".devX" when it is
+  dev: -1
+  # numpy will by default use the ABI feature level for the first numpy version
+  # that added support for the oldest currently-supported CPython version; see
+  # https://github.com/numpy/numpy/blob/v2.0.0rc1/numpy/_core/include/numpy/numpyconfig.h#L124
+  default_abi_level: 1.23
+  # ensure is_freethreading gets detected as a used variable
+  touch_is_freethreading: ${{ is_freethreading }}
+
+package:
+  name: numpy
+  version: ${{ version }}${{ dev if dev != -1 else "" }}
+
+source:
+  # The sdist distributed by numpy contains submodules; by taking the tarball
+  # from github we can everything pythran ourselves, but manually need to include
+  # the submodules (not in tarball due to dear-github/dear-github#214); for the
+  # list of modules see https://github.com/numpy/numpy/blob/main/.gitmodules
+  - url: https://github.com/numpy/numpy/archive/refs/tags/v${{ version }}.tar.gz
+    sha256: d040da6eb210b6e04cb76fa40b6ab221858ed9e188828ee8d8daa455cf41456f
+  # https://github.com/numpy/numpy/tree/v{{ version }}/numpy/_core/src
+  - git: https://github.com/google/highway.git
+    rev: 0b696633f9ad89497dd5532b55eaa01625ad71ca
+    target_directory: numpy/_core/src/highway
```

Comment:
Please keep the order as before: the submodules are sorted by their target directory, then the git remote, then the hash.

Reply:
done!